src/dom/extra.js

```diff
@@ -93,7 +93,7 @@ export const text = function(value) {
 export const val = function(value) {
 
   if(value === undefined) {
-    return this[0].value;
+    return this[0] ? this[0].value : undefined;
```

Comment:
Please check `this.length` (more consistent with rest of codebase)

Reply:
no problem